ui-terminal.c

```diff
@@ -269,25 +269,29 @@ static void ui_window_draw(Win *win) {
 	}
 }
 
-void ui_window_style_set(Ui *tui, int win_id, Cell *cell, enum UiStyle id) {
+void ui_window_style_set(Ui *tui, int win_id, Cell *cell, enum UiStyle id, bool keep) {
 	CellStyle set = tui->styles[win_id * UI_STYLE_MAX + id];
-
 	if (id != UI_STYLE_DEFAULT) {
-		set.fg = is_default_fg(set.fg)? cell->style.fg : set.fg;
-		set.bg = is_default_bg(set.bg)? cell->style.bg : set.bg;
+		CellStyle def = tui->styles[win_id * UI_STYLE_MAX + UI_STYLE_DEFAULT];
+		if (keep) {
+			set.fg = is_default_fg(cell->style.fg, def.fg) ? set.fg : cell->style.fg;
```

Comment:
```suggestion
			if (!is_default_fg(cell->style.fg) && !cell_color_equal(cell->style.fg, def.fg))
				set.fg = cell->style.fg;
```

These ternary operations are too difficult to follow with the proposed changes.